src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -120,6 +120,38 @@ class OMPStandaloneDirective(OMPDirective, StandaloneDirective):
     '''
 
 
+class OMPDeclareTargetDirective(OMPStandaloneDirective):
+    '''
+    Class representing an OpenMP Declare Target directive in the PSyIR.
+
+    '''
+    def gen_code(self, parent):
+        '''Generate the fortran OMP Declare Target Directive and any
+        associated code.
+
+        :param parent: the parent Node in the Schedule to which to add our \
+                       content.
+        :type parent: sub-class of :py:class:`psyclone.f2pygen.BaseGen`
+        '''
+        # Check the constraints are correct
+        self.validate_global_constraints()
+
+        # Generate the code for this Directive
+        parent.add(DirectiveGen(parent, "omp", "begin", "declare", "target"))
+
+    def begin_string(self):
```

Comment:
Again, could we give this Node Clauses instead?

Reply:
I am not sure about this one. Declare doesn't exist alone, and in fact the OpenMP standard speaks about the "declare target directive" with the clauses being "to" "link" and "device_type" https://www.openmp.org/spec-html/5.0/openmpsu62.html

Reply:
It's OK. I was thinking that with Clause nodes we would do away with `begin_string` altogether but of course we don't.